utils/FIS/FIS12/fisChecks.ts

```diff
@@ -289,3 +290,146 @@ export const validateContext = (context: any, msgIdSet: any, pastCall: any, cure
+          ) {
+            errorObj.cancellationFee = `Cancellation terms percentage or amount for ${descriptorCode} has changed`
+          }
+        }
+      }
+    } else {
+      errorObj.cancellationTerms = `cancellation_terms should be an array in /${action}`
+    }
+  } catch (error: any) {
+    logger.error(`!!Error while checking cancellation_terms in /${action}, ${error.stack}`)
+  }
+
+  return errorObj
+}
+
+export const validatePayments = (payments: any, action: string, quote: any) => {
+  try {
+    const errorObj: any = {}
+    const totalPaymentsAmount = payments
+      .filter((payment: any) => payment.params && payment.params.amount)
+      .reduce((total: any, payment: any) => total + parseFloat(payment.params.amount), 0)
+    const quotePriceValue = parseFloat(quote.price.value)
+
+    if (totalPaymentsAmount !== quotePriceValue) {
+      errorObj.paymentsAmountMismatch = `Total payments amount (${totalPaymentsAmount}) does not match with quote.price.value (${quotePriceValue})`
+    }
+
+    for (let i = 1; i < payments.length; i++) {
+      const payment = payments[i]
+
+      if (!payment.status) {
+        errorObj.payments = `status is missing in payments`
+      } else {
+        const allowedStatusValues = ['NOT_PAID', 'PAID', 'NOT-PAID']
+
+        if (!allowedStatusValues.includes(payment.status)) {
+          errorObj.paymentStatus = `Invalid value for status. It should be either of NOT_PAID or PAID.`
+        }
+      }
+
+      if (payment.time && payment.time.range) {
+        const { start, end } = payment.time.range
+        const startTime = new Date(start).getTime()
+        const endTime = new Date(end).getTime()
+
+        if (isNaN(startTime) || isNaN(endTime) || startTime > endTime) {
+          errorObj.invalidTimeRange = `Invalid time range for payment at index ${i}`
+        }
+
+        if (i > 0) {
+          const prevEndTime = new Date(payments[i - 1].time.range.end).getTime()
+          if (startTime <= prevEndTime) {
+            errorObj.timeRangeOrderError = `Time range order error for payment at index ${i}`
+          }
+        }
+      } else {
+        errorObj.missingTimeRange = `Missing time range for payment at index ${i}`
+      }
+    }
+
+    if (payments[0].tags) {
+      // Validate payment tags
+      const tagsValidation = validatePaymentTags(payments[0].tags)
+      if (!tagsValidation.isValid) {
+        Object.assign(errorObj, { tags: tagsValidation.errors })
+      }
+    }
+
+    if (!payments[0].collected_by) {
+      errorObj.payments = `collected_by  is missing in payments`
+    } else {
+      const allowedCollectedByValues = ['BPP', 'BAP']
+      const allowedStatusValues = ['NOT_PAID', 'PAID', 'NOT-PAID']
+
+      const collectedBy = getValue(`collected_by`)
+      if (collectedBy && collectedBy !== payments[0].collected_by) {
+        errorObj.collectedBy = `Collected_By didn't matched with what was send in previous call.`
```

Comment:
pls check grammatical errors: didn't match*.....what was sent*

Reply:
Fixed in latest commit